### PR TITLE
Only run tests on Linux - hddtemp and dmcache

### PR DESCRIPTION
**Pull request by rickard-von-essen**

### CODE REVIEW

File: plugins/inputs/hddtemp/go-hddtemp/hddtemp.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// +build linux
```
Comment (danielnelson) [inline plugins/inputs/hddtemp/go-hddtemp/hddtemp.go]: Why linux only?

File: plugins/inputs/dmcache/dmcache_linux_test.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// +build !linux
```
Comment (danielnelson) [inline plugins/inputs/dmcache/dmcache_linux_test.go]: This doesn't seem right, why are you removing test on linux and calling it dmcache_linux_test?

Comment (rickard-von-essen) [inline plugins/inputs/dmcache/dmcache_linux_test.go]: hmm, yeah that went a bit to fast.. I'll fix it.

File: plugins/inputs/hddtemp/go-hddtemp/hddtemp.go (hunk 1 above)
Comment (rickard-von-essen) [inline plugins/inputs/hddtemp/go-hddtemp/hddtemp.go]: Because hddtemp is Linux only but reading the code a bit it actually runs on any platform.

**Comment by rickard-von-essen** [comment]
Sorry that was sloppy. Now it should look better.

**Comment by danielnelson** [comment]
I think the hddtemp plugin incl tests should just be any platform.

**Comment by rickard-von-essen** [comment]
> I think the hddtemp plugin incl tests should just be any platform.

Yes, that is fine too.

**Pull request merged**